Edits proposed by Med (#8)
mirjak/draft-dhc-dhcpv4-over-dhcpv6-ra: draft-porfiri-dhc-dhcpv4-over-dhcpv6-ra.md

                 '-----------'             '-----------'

~~~
{: #fig_4o6RA title="Architecture Example with Legacy DHCP Client" artwork-align="center"}

This document specifies the encapsulation
and decapsulation described in {{RFC7341}} to be performed in the Relay Agent
whereas the DHCPv4 client does not require any change.
In this case it is up to the Relay Agent to provide the full DHCP
4o6 support whereas a legacy client is not aware of being served
via a DHCP 4o6 service.
All prerequisites and configuration that apply to the DHCP client in
{{Section 5 of RFC7341}} shall be applied to 4o6RA instead.

{{RFC7341}} specifies that before applying for an IPv4 address via a
DHCPv4-query message, the client must identify a suitable network interface
for the address. In the case described in this document where the
client functionality described in {{RFC7341}} is replaced by 4o6RA,

> it's 4o6RA that shall also identify a suitable interface, that can be

Author: ```suggestion
it is 4o6RA that shall also identify a suitable interface, that can be
```

Layer 2 Relay Agents receiving DHCPV4-QUERY or DHCPV4-RESPONSE messages
are expected to handle them as specified in {{Section 6 of RFC6221}}.

DHCPv6 servers must be compliant with 4o6 according to {{RFC7341}}.
No additional requirements on DHCPv6 servers are set by this specification.

## Intermediate relays

Intermediate relays shall behave according to section 10 of {{RFC7341}}.

## 4o6RA and Topology Discovery {#topology_considerations}

In some networks the configuration of a host may depend on the
topology.  However, when the new host attaches to a
network, it may be unaware of the topology and respectively how it
has to be configured.

DHCPv4 {{RFC2131}} and DHCPv6 {{RFC3315}} specifications
describe how addresses can be allocated to clients based on network
topology information provided by a DHCP relay, typically.

Address/prefix allocation decisions are integral to the allocation of
addresses and prefixes in DHCP. The argument is described in details
in {{RFC7969}}, here we want to guarantee that 4o6RA does not
break any legacy capability when related to the use of topology.

The topology discovery as described in {{RFC7969}} differs between

> IPv4 and IPv6 as in IPv4 it's only the first Relay Agent that can

Author: ```suggestion
IPv4 and IPv6 as in IPv4 it is only the first Relay Agent that can
```

> IPv4 and IPv6 as in IPv4 it is only the first Relay Agent that can

Author: ```suggestion
IPv4 and IPv6 because for IPv4 it is only the first Relay Agent that can
```

> set the giaddr field (section 3.1 of {{RFC7969}}) thus in a generic

Author: ```suggestion
set the giaddr field (section 3.1 of {{RFC7969}}). Thus in a generic
```

network involving more than one Relay Agent only part of the topology
is transported via DHCPv4.

> When in IPv6 context and using DHCPv6, all Relay Agents can fill
> link-address and Interface-ID options, that allows transporting
> to the DHCPv6 server the information about the complete path
> between the DHCPv6 client and the DHCPv6 server.
> When in a L2 network, Lightway DHCPv6 Relay Agents {{RFC6221}}
> can be used. Then, topology information for the given IP address
> can be obtained from the DHCPv6 server and used, for configuration
> or other purposes.

Author: ```suggestion
When in IPv6 context and using DHCPv6, all Relay Agents can send
link-address and Interface-ID options, that provide
information about the complete path
between the DHCPv6 client and the DHCPv6 server to the DHCPv6 server.
When in a L2 network, Lightway DHCPv6 Relay Agents {{RFC6221}}
can be used. Then, topology information for the given IP address
can be obtained from the DHCPv6 server and used for configuration
or other purposes.
```

The adoption of {{RFC7341}} at the client permits to use the DHCPv6

> mechanisms for topology discovery even in case of DHCPv4, because
> of the DHCPv6 encapsulation and because the DHCPv6 Relay Agent knows

Author: ```suggestion
mechanisms for topology discovery even in case of DHCPv4.
as using DHCPv6 encapsulation the DHCPv6 Relay Agent knows
```

{: #fig_4o6LDRA title="Topology path preserved with LDRA" artwork-align="center"}

The assumed architecture is shown in {{fig_4o6LDRA}} where the whole

> Relay Agent is built up with cooperating 4o6RA and LDRA, and an internal interface to

Author: ```suggestion
The Relay Agent is built up with cooperating 4o6RA and LDRA, and an internal interface to
```

          | L2 Network  |
 +--------+-+         +-+------+----------+
 |   DHCP   |         |  4o6   | DHCP 4o6 |
 |  Client  +---------+  Relay +  Server  |
 |  on CPE  |         |  Agent |          |
 +--------+-+         +-+------+----------+
          |             |
           '-----------'

~~~
{: #fig_4o6RAserver title="Topology path preserved 4o6 Relay Agent in DHCP server" artwork-align="center"}

# Deployment Considerations

As clients are not aware of the presence of 4o6RA, the network deployment needs to ensure that
all DHCPv4 broadcast and unicast messages from clients are steered to a 4o6RA.
This can, e.g., be achieved by placing the 4o6RA in a central position that can observe all traffic
from the clients or use of address translation with the 4o6RA address for unicast
messages.

# Security Considerations {#seccons}

This documents specifies the applicability of 4o6 DHCP in a scenario where legacy IPv4 clients are

> connected to 4o6 DHCP Relay Agent that performs the en- and decapsulation. This document

Author: ```suggestion
connected to 4o6 DHCP Relay Agents that performs the encapsulation and decapsulation. This document
```

does not change anything else in the 4o6 DHCP specification and therefore the
security consideration of {{RFC7341}} still apply.

This mechanism differs from {{RFC7341}} as the DHCP client
sends and receives DHCPv4 messages, whereas in {{RFC7341}} it
only sends DHCPv6 messages. This makes it possible that
DHCPv4 messages could reach a DHCPv4 server without using the 4o6RA.
While this can cause erroneous state in both clients and servers
and potentially even lead to misconfigurations that impact reachability,

> this is not seen as a security concern rather than a deployment error
> and even though it may be used for attacks from within the network,
> it's not new and it's not being introduced because of this specification.

Author: ```suggestion
this is not seen as a security concern rather than a deployment error.
Further, even though it may be used for attacks from within the network,
this is not new and it is not introduced because of this specification.
```

# Example Use Case: Topology Discovery for IPv4-only Radio Unit in the RAN Switched Fronthaul {#usecase}

The Radio Fronthaul Network (FH) is built up with Radio Units (RU) and
Baseband Units (BB), each being an IP host.

> Each RU is unique as it's tied to a set of antennas, each antennas

Author: ```suggestion
Each RU is unique as it is tied to a set of antennas, and each antenna
```

is serving a specific Cell and Sector.

> RU is configured by BB depending to wich Cell and Sectors it serves,
> that dependency is only specified by the cabling between RU and antennas.

Author: ```suggestion
Each RU is configured by the BB depending on the Cell and Sectors it serves.
However, that dependency is only specified by the cabling between RU and antennas.
```

> In the legacy, BB is direcly cabled to a set of RUs so that the
> BB can recognize the relationship between RU and Cell/Sectors
> by relating the cabling towards the RU and the information
> about cabling of RU and antennas.

Author: ```suggestion
If BB is directly cabled to a set of RUs, the
BB can recognize the relationship between Russia and Cell/Sectors
based on the cabling between the RUs and antennas.
```

> The introduction of a switched network between RUs and BBs has
> added a level of complexity that requires the BBs to have a deeper
> knowledge of the topology, involving all the cabling between
> the RUs and the switched network in order to properly configure the RUs.

Author: ```suggestion
The introduction of a switched network between RUs and BBs has
added a level of complexity that requires the BBs to have a deeper
knowledge of the topology in order to properly configure the RUs,
involving knowledge all the cabling in the switched network.
```

> The switched network can be generic and the level of complexity can
> be related to the examples described in section 3 of {{RFC7969}}.

Author: ```suggestion
Examples for switched networks are show in section 3 of {{RFC7969}}
and demonstrate the different levels of complexity.
```

> An example of FH is depicted in {{l2_switched_fh}}.

Author: ```suggestion
An example of a FH is depicted in {{l2_switched_fh}}.
```

~~~
{: #l2_switched_fh title="Layer 2 Switched Fronthaul Example" artwork-align="center"}

Among the various alternatives, DHCP topology knowledge can be used

> for solving the RU configuration problem when FH is IPv6. Such solution

Author: ```suggestion
for solving the RU configuration problem when the FH is IPv6. Such solution
```

would use the topology discovery mechanisms described in section 3.2

> of {{RFC7969}}. The same mechanisms applies when RU are IPv4 and

Author: ```suggestion
of {{RFC7969}}. The same mechanisms applies when RUs are connected via IPv4 and
```

implement 4o6 according to {{RFC7341}}.

In order to extend the solution described above also to the case where

> RU is IPv4 but it cannot support {{RFC7341}}, it's possible to adopt

Author: ```suggestion
RUs are using IPv4 but cannot support {{RFC7341}}, it is possible to adopt
```

> the mechanisms described in this document by introducing 4o6RA at

Author: ```suggestion
the mechanisms described in this document by introducing 4o6RA in
```
